tools/pkglint/lib/rules.ts

```diff
@@ -218,14 +219,41 @@ function cdkModuleName(name: string) {
     name = name.replace(/^aws-cdk-/, '');
     name = name.replace(/^@aws-cdk\//, '');
 
+    const dotnetSuffix = name.split('-').map(s => s === 'aws' ? 'AWS' : caseUtils.pascal(s)).join('.');
+
     return {
         javaPackage: `software.amazon.awscdk${isCdkPkg ? '' : `.${name.replace(/^aws-/, 'services-').replace(/-/g, '.')}`}`,
         mavenArtifactId: isCdkPkg ? 'cdk'
                                   : name.startsWith('aws-') ? name.replace(/^aws-/, '')
-                                                            : `cdk-${name}`
+                                                            : `cdk-${name}`,
+        dotnetNamespace: `Amazon.CDK${isCdkPkg ? '' : `.${dotnetSuffix}`}`
     };
 }
 
+/**
+ * JSII .NET namespace is required and must look sane
```

Comment:
Can this not be optional (with a sane default derived), have to look sane only if present?

Reply:
We could make `packageId` *or* `namespace` optional, but not both. If both were optional and jsii tried to determine a sane default, it would be something like `Aws.Cdk.Aws.ApiGateway`, rather than `Amazon.CDK.AWS.ApiGateway`.

Reply:
Discussion moved here: https://github.com/awslabs/jsii/issues/198